blocks/api/utils.js

```diff
@@ -72,7 +72,20 @@ export function normalizeIconObject( icon ) {
 
 	if ( icon.background ) {
 		const tinyBgColor = tinycolor( icon.background );
-		if ( ! icon.foreground ) {
+		let shouldComputeForeground = true;
+		if ( icon.foreground ) {
+			const tinyFgColor = tinycolor( icon.foreground );
+			if ( tinycolor.isReadable(
+				tinyBgColor,
+				tinyFgColor,
+				{ level: 'AA', size: 'large' }
+			) ) {
+				shouldComputeForeground = false;
+			} else if ( window ) {
+				window.console.warn( `The icon background color ${ icon.background } and the foreground color ${ icon.foreground } are not readable together. The foreground color was ignored, and an automatic foreground color computation was used.` );
```

Comment:
This is about contrast, right? We should explain *why* the colours aren't readable together so anyone seeing this error understands it. Maybe `The icon's foreground color ${ icon.foreground } does not contrast enough against its background color ${ icon.background }. The selected foreground color was ignored, and an automatic foreground color computation was used instead.`